blaze/mahimahi/server/server.py

```diff
@@ -136,36 +143,21 @@ def start_server(
 
                 backup_file_body = file.body
                 try:
-                    if extract_critical_requests:
-                        # if this is an html file, then we want to insert our snippet here
-                        # to extract critical requests.
-                        if file.headers.get("content-type", "") == "text/html":
-                            uncompressed_body = file.body
-                            gzipped_file = False
-                            if (
-                                file.headers is not None
-                                and file.headers.get("content-encoding") is not None
-                                and "gzip" in file.headers.get("content-encoding")
-                            ):
-                                gzipped_file = True
-                                uncompressed_body = gzip.GzipFile(fileobj=BytesIO(uncompressed_body)).read()
-                            uncompressed_body = prepend_javascript_snippet(uncompressed_body)
-                            if gzipped_file:
-                                out = BytesIO()
-                                with gzip.GzipFile(fileobj=out, mode="wb") as f:
-                                    f.write(uncompressed_body.encode())
-                                file.body = out.getvalue()
-                            else:
-                                file.body = uncompressed_body
-                    # Save the file's body to file
+                    if extract_critical_requests and "text/html" in file.headers.get("content-type", ""):
+                        file.body = inject_extract_critical_requests_javascript(file)
                     file_path = os.path.join(file_dir, file.file_name)
                     with open(os.open(file_path, os.O_CREAT | os.O_WRONLY, 0o644), "wb") as f:
                         f.write(file.body)
                 except TypeError as e:
-                    # file.body somehow because corrupted. happens in nbcnews where some ad server file is of type html
-                    # # this messes up the encoding
+                    file_path = os.path.join(file_dir, file.file_name)
+                    with open(os.open(file_path, os.O_CREAT | os.O_WRONLY, 0o644), mode="w", encoding="utf8") as f:
+                        f.write(file.body)
```

Comment:
these two code blocks are the same -- can pull it out of the try/except

Reply:
the writing mode is different - if we cannot write it as bytes, then we get a TypeError and try writing as a string

Reply:
its fine then